### PR TITLE
Ckpt fixes pytorch update

### PR DESCRIPTION
**Pull request by chtruong814**
# What does this PR do ?

Fixes additional tests for the pytorch container 25.01 bump.

* [L2_PTQ_Llama2_FP8](https://github.com/NVIDIA/NeMo/actions/runs/13394336539/job/37411726892)
* [L2_NeMo_2_PTQ_Llama2_FP8](https://github.com/NVIDIA/NeMo/actions/runs/13394336539/job/37411728719)
* [L2_NeMo_2_SSM_Finetuning](https://github.com/NVIDIA/NeMo/actions/runs/13394336539/job/37411728189)
* [L2_NMT_Attention_is_All_You_Need_Finetuning](https://github.com/NVIDIA/NeMo/actions/runs/13394336539/job/37411727652)

The PTQ and SSM finetuning tests were failing because we had to set the `ckpt_load_strictness` to `log_all` similar to what we've done in several cases already. The NMT test was failing because the data class was not using the default factory for a mutable field, which is required for the python update to 3.12.

This is merging into @ko3n1g 's Pytorch bump branch.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Fix additional tests due to Pytorch update

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by chtruong814** [comment]
Not ready for review yet.  Wanting to run the tests against some of this and will need the respective devs to take a look even if this passes.

**Review by dimapihtar — APPROVED** [review]
LGTM. Thank you!
